src/KeyboardInputSchemes.js

```diff
@@ -0,0 +1,128 @@
+        refreshScene: { code: "KeyR", altKey: true},
+        showHide: { key: "?" },
+        stopProgram: { code: "KeyS", altKey: true},
+        toggleAnnouncements: { code: "KeyA", altKey: true}
+    },
+    "nvda": {
+        addCommandToBeginning: { code: "KeyB", altKey: true, ctrlKey: true},
+        addCommandToEnd: { code: "KeyE", altKey: true, ctrlKey: true},
+        announceScene: { code: "KeyI", altKey: true, ctrlKey: true},
+        decreaseProgramSpeed: { key: "<" },
+        increaseProgramSpeed: { key: ">" },
+        playPauseProgram: { code: "KeyP", altKey: true, ctrlKey: true},
+        refreshScene: { code: "KeyR", altKey: true, ctrlKey: true},
+        showHide: { key: "?" },
+        stopProgram: { code: "KeyS", altKey: true, ctrlKey: true},
+        toggleAnnouncements: { code: "KeyA", altKey: true, ctrlKey: true}
+    }
+};
+
+const labelMessageKeysByCode = {
+    "KeyA": "KeyboardInputModal.KeyLabels.A",
+    "KeyB": "KeyboardInputModal.KeyLabels.B",
+    "KeyE": "KeyboardInputModal.KeyLabels.E",
+    "KeyI": "KeyboardInputModal.KeyLabels.I",
+    "KeyP": "KeyboardInputModal.KeyLabels.P",
+    "KeyS": "KeyboardInputModal.KeyLabels.S",
+    "KeyR": "KeyboardInputModal.KeyLabels.R"
+};
+
+const labelMessageKeysByKey = {
+    "?": "KeyboardInputModal.KeyLabels.QuestionMark",
+    ">": "KeyboardInputModal.KeyLabels.GreaterThan",
+    "<": "KeyboardInputModal.KeyLabels.LessThan"
+};
+
+export function getLabelMessageKeyFromKeyDef (keyDef: KeyDef) {
+    if (keyDef.code && labelMessageKeysByCode[keyDef.code]) {
+        return labelMessageKeysByCode[keyDef.code];
+    }
+    else if (keyDef.key && labelMessageKeysByKey[keyDef.key]) {
+        return labelMessageKeysByKey[keyDef.key];
+    }
+    else {
+        return "";
+    }
+};
+
+const iconMessageKeysByCode = {
+    "KeyA": "KeyboardInputModal.KeyIcons.A",
+    "KeyB": "KeyboardInputModal.KeyIcons.B",
+    "KeyE": "KeyboardInputModal.KeyIcons.E",
+    "KeyI": "KeyboardInputModal.KeyIcons.I",
+    "KeyP": "KeyboardInputModal.KeyIcons.P",
+    "KeyS": "KeyboardInputModal.KeyIcons.S",
+    "KeyR": "KeyboardInputModal.KeyIcons.R"
+};
+
+const iconMessageKeysByKey = {
+    "?": "KeyboardInputModal.KeyIcons.QuestionMark",
+    ">": "KeyboardInputModal.KeyIcons.GreaterThan",
+    "<": "KeyboardInputModal.KeyIcons.LessThan"
+};
+
+export function getIconMessageKeyFromKeyDef (keyDef: KeyDef) {
+    if (keyDef.code && iconMessageKeysByCode[keyDef.code]) {
+        return iconMessageKeysByCode[keyDef.code];
+    }
+    else if (keyDef.key && iconMessageKeysByKey[keyDef.key]) {
+        return iconMessageKeysByKey[keyDef.key];
+    }
+    else {
+        return "";
+    }
+};
+
+export function keyboardEventMatchesKeyDef (e: KeyboardEvent, keyDef: KeyDef) {
+    if (e.code === keyDef.code || e.key === keyDef.key) {
+        if (!!(keyDef.altKey) !== !!(e.altKey)) {
+            return false;
+        }
+        if (!!(keyDef.shiftKey) !== !!(e.shiftKey)) {
```

Comment:
Any key bindings with shift key gets filtered by this check, because keyDef.shiftKey is undefined for "<", ">" and "?", which result in those key bindings to not work. I was thinking maybe we could make keyDefs with code + shiftKey combination?

Reply:
I agree, and this explains recent issues getting the bindings to work as well. When I added tests, I hardened the checks to fix false matches (triggering actions when the right keys weren't hit). That seems to have resulted in the behaviour you picked up on.

Reply:
I ended up relaxing this check, my reasoning is that since caps lock is often used as a screenreader activation key, we may not want to be as strict about ctrl-a vs ctrl-A.   We should discuss in the meeting, it may make more sense not to include / rely on the shift state at all in key bindings.